### PR TITLE
Improve overlay scaling

### LINKED ISSUE #24: Improve AmbassadorList responsiveness
Could we make the list a bit taller in smaller viewports? According to the twitch dev docs [1] the overlay can be up to 5rem from top/bottom. Another idea would be to move the scroll arrow button in front of the list to gain a few more pixels. I can make a pull request if you would consider it.

Current:
Overlay at 1024 pixel width

Proposed arrow button above list:
Mockup arrow button above list

------------------

1) https://dev.twitch.tv/docs/extensions/designing/#video-overlay-extensions

### PR DESCRIPTION
**Pull request by pjeweb**
My first attempt on the improved scaling.

- Instead of scaling everything with `scale()` i opted to scale the list components separately with some `calc(x * clamp())` math to keep the scaling within bounds. With that we can make the list always use the full height of the overlay container which i have positioned to keep the recommended 5rem (=50px) space above and below.
- I moved the arrows above the list. I am not sure if you actually wanted that
- I added a `mask-image` to fade out the ambassador list
- I kept the previous scaling method for the card (using scale transforms) but added some more steps

Resolves #24

### CODE REVIEW

File: src/pages/overlay/index.scss
Comment (MattIPv4) [inline src/pages/overlay/index.scss:14]: Curious why this is being defined as a CSS var, rather than a SCSS var? Maybe personal pref, but with variables already defined in SCSS, I'd probably lean on a consistent approach to them?

Comment (pjeweb) [reply src/pages/overlay/index.scss:14]: Good catch. I tried to use media queries first to make the css variable dynamic (in steps), but moved to the clamp solution without media queries. I will update the PR later.

**Review by abdullahmorrison — CHANGES_REQUESTED** [review]
The top of the overlay gets blocked by the Twitch header. It needs to be moved down a bit.

image

**Comment by abdullahmorrison** [comment]
I like the arrow design 👍🏾

**Comment by pjeweb** [comment]
> The top of the overlay gets blocked by the Twitch header. It needs to be moved down a bit.

Yeah I saw that when testing as well. The problem is that the title can be even bigger and will wrap to multiple (many) lines if the window is very small. And this only shows up in theatre mode or if embedded, so I just ignored that. But if you would prefer, I can move it down a bit. That is very easy to change with these changes.

Screenshot 2023-01-23 at 15 05 59

**Comment by abdullahmorrison** [comment]
ye, on the extreme case it's fine, I just want it below the title if we assume there is not wrapping

File: src/pages/overlay/components/activationButtons/activationButtons.module.scss
Comment (MattIPv4) [inline src/pages/overlay/components/activationButtons/activationButtons.module.scss:4]: Can this adjusted margin be copied over to overlaysettings to make sure they stay aligned?

Comment (pjeweb) [reply src/pages/overlay/components/activationButtons/activationButtons.module.scss:4]: yes sure. it is aligned at the top now, but otherwise it does not scale. i guess thats fine? the toggle button should not get any smaller, right?

Comment (MattIPv4) [reply src/pages/overlay/components/activationButtons/activationButtons.module.scss:4]: Thanks! That seems fine to me, I can't imagine scaling it will be useful (maybe at a certain point it should disappear?) If more settings are added down the road maybe it collapses into another menu like the list, but that's a future problem.

**Comment by pjeweb** [comment]
PR is rebased and should leave enough space for the header, settings are aligned at the top.

Screenshot 2023-01-23 at 21 33 16

**Comment by pjeweb** [comment]
Should I rebase this PR again or should I wait until you are sure you want to merge it?

**Comment by abdullahmorrison** [comment]
I'd say wait for me to review it

File: src/pages/overlay/components/activationButtons/activationButtons.module.scss
Comment (abdullahmorrison) [inline src/pages/overlay/components/activationButtons/activationButtons.module.scss:8]: can you make the size of the button bigger. it feels a bit too small.

**Review by abdullahmorrison — CHANGES_REQUESTED** [review]
The card information is to compacted. If we assume that the info for the ambassador story and conservation mission is full, there needs to be around 5px of margin between each row of the card. Also 5px at the bottom of the card.

image

image

File: src/utils/compositions/ambassador/ambassador.module.scss
Comment (pjeweb) [reply src/utils/compositions/ambassador/ambassador.module.scss]: the ambassador module does not get scaled using scale() anymore. to make the padding scale it must be relative or use the base size.  because the ambassador composition is being used for both overlay and panel, I do not use the base size but a relative size.

File: src/pages/panel/components/ambassadorPanel/AmbassadorPanel.tsx
Comment (pjeweb) [reply src/pages/panel/components/ambassadorPanel/AmbassadorPanel.tsx]: yeah the `ClassName` prop was added since the PR, can use that!

File: src/pages/overlay/components/activationButtons/activationButtons.module.scss
Comment (pjeweb) [reply src/pages/overlay/components/activationButtons/activationButtons.module.scss]: its original size was not square. it was 65 x 71. which size would you want?

File: src/pages/overlay/components/activationButtons/activationButtons.module.scss
Comment (pjeweb) [reply src/pages/overlay/components/activationButtons/activationButtons.module.scss]: see https://github.com/abdullahmorrison/AlveusTwitchExtension/pull/26#discussion_r1088469243

File: src/pages/overlay/components/activationButtons/activationButtons.module.scss
Comment (pjeweb) [reply src/pages/overlay/components/activationButtons/activationButtons.module.scss]: It is 65x65 for now.

**Comment by pjeweb** [comment]
Should all be changed as requested. I would hope we can merge it and work on it in separate PR/commits if necessary because the rebasing is very time consuming and error prone, as this PR touches so many files.

**Review by abdullahmorrison — APPROVED** [review]

**Pull request merged**
